types/dec_coin_test.go

```diff
@@ -4,80 +4,92 @@ import (
 	"strings"
 	"testing"
 
-	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
-func TestNewDecCoin(t *testing.T) {
-	require.NotPanics(t, func() {
+type decCoinTestSuite struct {
+	suite.Suite
+}
+
+func NewDecCoinTestSuite(t *testing.T) {
+	suite.Run(t, new(decCoinTestSuite))
+}
+
+func (s *decCoinTestSuite) SetupSuite() {
+	s.T().Parallel()
+}
+
+func (s *decCoinTestSuite) TestNewDecCoin() {
+	s.Require().NotPanics(func() {
 		sdk.NewInt64DecCoin(testDenom1, 5)
 	})
-	require.NotPanics(t, func() {
+	s.Require().NotPanics(func() {
 		sdk.NewInt64DecCoin(testDenom1, 0)
 	})
-	require.NotPanics(t, func() {
+	s.Require().NotPanics(func() {
 		sdk.NewInt64DecCoin(strings.ToUpper(testDenom1), 5)
 	})
-	require.Panics(t, func() {
+	s.Require().Panics(func() {
 		sdk.NewInt64DecCoin(testDenom1, -5)
 	})
 }
 
-func TestNewDecCoinFromDec(t *testing.T) {
-	require.NotPanics(t, func() {
+func (s *decCoinTestSuite) TestNewDecCoinFromDec() {
+	s.Require().NotPanics(func() {
 		sdk.NewDecCoinFromDec(testDenom1, sdk.NewDec(5))
 	})
-	require.NotPanics(t, func() {
+	s.Require().NotPanics(func() {
 		sdk.NewDecCoinFromDec(testDenom1, sdk.ZeroDec())
 	})
-	require.NotPanics(t, func() {
+	s.Require().NotPanics(func() {
 		sdk.NewDecCoinFromDec(strings.ToUpper(testDenom1), sdk.NewDec(5))
 	})
-	require.Panics(t, func() {
+	s.Require().Panics(func() {
 		sdk.NewDecCoinFromDec(testDenom1, sdk.NewDec(-5))
 	})
 }
 
-func TestNewDecCoinFromCoin(t *testing.T) {
-	require.NotPanics(t, func() {
+func (s *decCoinTestSuite) TestNewDecCoinFromCoin() {
+	s.Require().NotPanics(func() {
 		sdk.NewDecCoinFromCoin(sdk.Coin{testDenom1, sdk.NewInt(5)})
 	})
-	require.NotPanics(t, func() {
+	s.Require().NotPanics(func() {
 		sdk.NewDecCoinFromCoin(sdk.Coin{testDenom1, sdk.NewInt(0)})
 	})
-	require.NotPanics(t, func() {
+	s.Require().NotPanics(func() {
 		sdk.NewDecCoinFromCoin(sdk.Coin{strings.ToUpper(testDenom1), sdk.NewInt(5)})
 	})
-	require.Panics(t, func() {
+	s.Require().Panics(func() {
 		sdk.NewDecCoinFromCoin(sdk.Coin{testDenom1, sdk.NewInt(-5)})
 	})
 }
 
-func TestDecCoinIsPositive(t *testing.T) {
+func (s *decCoinTestSuite) TestDecCoinIsPositive() {
 	dc := sdk.NewInt64DecCoin(testDenom1, 5)
-	require.True(t, dc.IsPositive())
+	s.Require().True(dc.IsPositive())
 
 	dc = sdk.NewInt64DecCoin(testDenom1, 0)
-	require.False(t, dc.IsPositive())
+	s.Require().False(dc.IsPositive())
 }
 
-func TestAddDecCoin(t *testing.T) {
+func (s *decCoinTestSuite) TestAddDecCoin() {
 	decCoinA1 := sdk.NewDecCoinFromDec(testDenom1, sdk.NewDecWithPrec(11, 1))
 	decCoinA2 := sdk.NewDecCoinFromDec(testDenom1, sdk.NewDecWithPrec(22, 1))
 	decCoinB1 := sdk.NewDecCoinFromDec(testDenom2, sdk.NewDecWithPrec(11, 1))
 
 	// regular add
 	res := decCoinA1.Add(decCoinA1)
-	require.Equal(t, decCoinA2, res, "sum of coins is incorrect")
+	s.Require().Equal(decCoinA2, res, "sum of coins is incorrect")
```

Comment:
we could do at the beginning of the function:

```
var require = s.Require()
```
And avoid other changes.

Reply:
We wouldn't avoid these changes anyway, diffs would just look different:

```diff
-	require.Equal(t, decCoinA2, res, "sum of coins is incorrect")
+	require.Equal(decCoinA2, res, "sum of coins is incorrect")
```